src/run.rs

```diff
@@ -135,7 +139,13 @@ impl WorkerSession {
             }
             Err(err) if self.trial_idx >= self.context.max_retries_per_op => Err(err),
             Err(err) if self.context.should_stop() => Err(err),
-            Err(_) => {
+            Err(err) => {
+                let countdown = self.context.retry_countdown.fetch_sub(1, Ordering::Relaxed);
```

Comment:
Yes, it will just wrap around. This behavior is defined in the documentation: https://doc.rust-lang.org/std/sync/atomic/struct.AtomicU64.html#method.fetch_sub